internal/controllers/clusterextension_controller.go

```diff
@@ -157,15 +178,31 @@ func checkForUnexpectedFieldChange(a, b ocv1alpha1.ClusterExtension) bool {
 //
 //nolint:unparam
 func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.ClusterExtension) (ctrl.Result, error) {
+	l := log.FromContext(ctx).WithName("operator-controller")
 	// run resolution
 	bundle, err := r.resolve(ctx, *ext)
-	if err != nil {
-		// set right statuses
-		ext.Status.ResolvedBundle = nil
-		ext.Status.InstalledBundle = nil
+	if err != nil || bundle == nil {
+		l.V(1).Info("bundle resolve error: %v\n", err)
+
+		errorMessage := err.Error()
+		packageNotFoundPattern := regexp.MustCompile(`^no package "([^"]+)" found$`)
```

Comment:
likely we should get more of a type wrapping the error than do this

Reply:
Can we improve the errors on the resolution instead of this? That would probably be easier, given the resolution code is in the same file.

Reply:
yeah we should, mostly wanted to see if custom casing was the way to go

Reply:
seems like if the errs are changes to be different we can case on them much more reliably